packages/lib/scripts/compile_validator.js

```diff
@@ -22,7 +22,11 @@ const ajv = new Ajv({
 });
 const validate = ajv.compile(Message.jsonSchema);
 let moduleCode = standaloneCode(ajv, validate);
-fs.outputFileSync(
-	path.join(__dirname, "../dist/src/data/message_validate.js"),
-	`// generated by scripts/compile_validator.js\n// @ts-nocheck\n/* eslint-disable */\n${moduleCode}`
-);
+function write(output) {
+	fs.outputFileSync(
+		path.join(__dirname, output),
+		`// generated by scripts/compile_validator.js\n// @ts-nocheck\n/* eslint-disable */\n${moduleCode}`
+	);
+}
+write("../dist/src/data/message_validate.js");
+write("../src/data/message_validate.js");
```

Comment:
Where is `src/data/message_validate.js` used? From what I see tsc doesn't touch it because only `ts`, `tsx` and `d.ts` files are included. I also don't see any files directly requiring this file from the src dir.

Reply:
It's required from `src/data/messages_core.ts`. It can't currently be included by `tsc` as that would cause a circular dependency loop when trying to create the file.